internal/backend/remote-state/azure/auth/cli_auth.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package auth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+)
+
+type AzureCLIAuthConfig struct {
+	CLIAuthEnabled bool
+}
+
+type azureCLICredentialAuth struct{}
+
+func (cred *azureCLICredentialAuth) Name() string {
+	return "Azure CLI Auth"
+}
+
+func (cred *azureCLICredentialAuth) Construct(_ context.Context, config *Config) (azcore.TokenCredential, error) {
+	// The SubscriptionID and TenantID can be empty, and the logic of this will still be okay
+	return azidentity.NewAzureCLICredential(&azidentity.AzureCLICredentialOptions{
+		Subscription: config.StorageAddresses.SubscriptionID,
+		TenantID:     config.StorageAddresses.TenantID,
+	})
+}
+func (cred *azureCLICredentialAuth) Validate(config *Config) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+	if !config.CLIAuthEnabled {
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Azure Backend: Command Line credentials",
+			"You set use_cli to false, which prevents the CLI Auth from being used. Remove this unless you're testing internals or otherwise explicitly want to prevent CLI Authentication.",
```

Comment:
Suggestion/Nit:
"Use of command line auth (az) has been prevented by setting use_cli to false."